packages/ai/core/tool/mcp/types.ts

```diff
@@ -42,11 +40,14 @@ export type McpToolSet<TOOL_SCHEMAS extends ToolSchemas = 'automatic'> =
         };
       };
 
-export interface McpStdioServerConfig {
+/**
+ * @deprecated Use custom transports instead. Import stdio transport from 'ai/mcp-stdio' or implement your own transport using @modelcontextprotocol/sdk.
+ */
+interface McpStdioServerConfig {
   command: string;
   args?: string[];
   env?: Record<string, string>;
-  stderr?: IOType | Stream | number;
+  stderr?: any | number;
   cwd?: string;
   type: 'stdio';
 }
```

Comment:
we might need to make a hard breaking change and just remove it (fine bc things are experimental)